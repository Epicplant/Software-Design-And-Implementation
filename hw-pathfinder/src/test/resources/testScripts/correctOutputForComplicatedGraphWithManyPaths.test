# Tests that correct path and minimum cost is displayed for a more complicated graph with many paths

# Creates a new graph
CreateGraph newGraph

# Adds the node A to newGraph
AddNode newGraph A

# Adds the node B to newGraph
AddNode newGraph B

# Adds the node C to newGraph
AddNode newGraph C

# Adds the node D to newGraph
AddNode newGraph D

# Adds the node E to newGraph
AddNode newGraph E

# Adds the node F to newGraph
AddNode newGraph F

# Adds the node G to newGraph
AddNode newGraph G

# Adds the node H to newGraph
AddNode newGraph H

# Adds the node I to newGraph
AddNode newGraph I

# Adds the node J to newGraph
AddNode newGraph J

# Adds edge of weight 1 to the graph
AddEdge newGraph D B 1

# Adds edge of weight 9 to the graph
AddEdge newGraph D A 9

# Adds edge of weight 1 to the graph
AddEdge newGraph D B 1

# Adds edge of weight 2 to the graph
AddEdge newGraph B G 2

# Adds edge of weight 7 to the graph
AddEdge newGraph B H 7

# Adds edge of weight 6 to the graph
AddEdge newGraph B C 6

# Adds edge of weight 12 to the graph
AddEdge newGraph G A 12

# Adds edge of weight 10 to the graph
AddEdge newGraph G C 10

# Adds edge of weight 5 to the graph
AddEdge newGraph A C 5

# Adds edge of weight 3 to the graph
AddEdge newGraph C H 3

# Adds edge of weight 7 to the graph
AddEdge newGraph C F 7

# Adds edge of weight 8 to the graph
AddEdge newGraph H F 8

# Adds edge of weight 3 to the graph
AddEdge newGraph H C 3

# Adds edge of weight 13 to the graph
AddEdge newGraph F I 13

# Adds edge of weight 11 to the graph
AddEdge newGraph I E 11

# Adds edge of weight 9 to the graph
AddEdge newGraph I J 9

# Adds edge of weight 2 to the graph
AddEdge newGraph E J 2

# Find the least weight path from D to J
FindPath newGraph D J