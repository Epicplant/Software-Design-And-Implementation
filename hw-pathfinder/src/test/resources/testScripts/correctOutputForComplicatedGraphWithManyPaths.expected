# Tests that correct path and minimum cost is displayed for a more complicated graph with many paths

# Creates a new graph
created graph newGraph

# Adds the node A to newGraph
added node A to newGraph

# Adds the node B to newGraph
added node B to newGraph

# Adds the node C to newGraph
added node C to newGraph

# Adds the node D to newGraph
added node D to newGraph

# Adds the node E to newGraph
added node E to newGraph

# Adds the node F to newGraph
added node F to newGraph

# Adds the node G to newGraph
added node G to newGraph

# Adds the node H to newGraph
added node H to newGraph

# Adds the node I to newGraph
added node I to newGraph

# Adds the node J to newGraph
added node J to newGraph

# Adds edge of weight 1 to the graph
added edge 1.000 from D to B in newGraph

# Adds edge of weight 9 to the graph
added edge 9.000 from D to A in newGraph

# Adds edge of weight 1 to the graph
added edge 1.000 from D to B in newGraph

# Adds edge of weight 2 to the graph
added edge 2.000 from B to G in newGraph

# Adds edge of weight 7 to the graph
added edge 7.000 from B to H in newGraph

# Adds edge of weight 6 to the graph
added edge 6.000 from B to C in newGraph

# Adds edge of weight 12 to the graph
added edge 12.000 from G to A in newGraph

# Adds edge of weight 10 to the graph
added edge 10.000 from G to C in newGraph

# Adds edge of weight 5 to the graph
added edge 5.000 from A to C in newGraph

# Adds edge of weight 3 to the graph
added edge 3.000 from C to H in newGraph

# Adds edge of weight 7 to the graph
added edge 7.000 from C to F in newGraph

# Adds edge of weight 8 to the graph
added edge 8.000 from H to F in newGraph

# Adds edge of weight 3 to the graph
added edge 3.000 from H to C in newGraph

# Adds edge of weight 13 to the graph
added edge 13.000 from F to I in newGraph

# Adds edge of weight 11 to the graph
added edge 11.000 from I to E in newGraph

# Adds edge of weight 9 to the graph
added edge 9.000 from I to J in newGraph

# Adds edge of weight 2 to the graph
added edge 2.000 from E to J in newGraph

# Find the least weight path from D to J
path from D to J:
D to B with weight 1.000
B to C with weight 6.000
C to F with weight 7.000
F to I with weight 13.000
I to J with weight 9.000
total cost: 36.000